### ISSUE TITLE
increase default timeout for ldap connections

### ISSUE THREAD

**Issue by aemneina** [rancher/rancher#8944]
The default of 1 second is aggressive for some environments. Bumping the default value will help prevent hangups on the ldap connection, which result in corrupted threads in the connection pool. <- probably need to fix that too.

**Comment by aemneina** [comment]
The symptoms of this issue are that when switching environments, you could hit these busted threads and you get immediately logged out.

**Comment by aemneina** [comment]
related to: https://github.com/rancher/rancher/issues/8945

**Comment by deniseschannon** [comment]
@mrajashree With the port from AD to the external auth microservice, what is the new default timeout?

**Comment by mrajashree** [comment]
@deniseschannon it's 5 seconds

**Comment by sangeethah** [comment]
@mrajashree since only AD is getting ported to the  external auth microservice , openLDAP will continue to have 1 second connection timeout?

**Comment by djk** [comment]
Could you make the LDAP timeout adjustable? 

I know in the DB table "settings" there is "api.auth.ldap.openldap.connection.timeout" which is set to "1000" which is presumably a second. It would be nice to expose this in the GUI.

**Comment by sangeethah** [comment]
Tested with rancher-server version v 1.6.6-rc4
Have AD auth enabled on the setup.
From within the rancher-server container, block the connection to AD ip address.

Try to launch the "Manage Environment" page that does user look up calls ,  projectmemebers api call take ~1 mt to return. Should this be only 5 secs ?

**Comment by prachidamle** [comment]
Can we re-run this test with TRACE on rancher/server? That should hopefully show us the exact time when a ldap failure is received in logs.

**Comment by deniseschannon** [comment]
Available in v1.6.8-rc1

**Comment by sangeethah** [comment]
Tested with rancher server version - v1.6.8-rc1

Have AD auth enabled on the setup.
From within the rancher-server container, block the connection to AD ip address.

Try to launch the "Manage Environment" page that does user look up calls , projectmemebers api call take ~1 mt to return.

Debug logs:
```
time="2017-08-11T21:20:16Z" level=debug msg="SearchIdentities by externalID: CN=Test User1,CN=Users,DC=tad,DC=rancher,DC=io and externalIDType: ldap_user" 
time="2017-08-11T21:20:16Z" level=debug msg="Failed to determine if object is type: ldap_user" 
time="2017-08-11T21:20:16Z" level=debug msg="Failed to determine if object is type: person" 
time="2017-08-11T21:20:16Z" level=debug msg="Query for GetIdentity(CN=Test User1,CN=Users,DC=tad,DC=rancher,DC=io): (objectClass=*)" 
time="2017-08-11T21:20:16Z" level=debug msg="Now creating Ldap connection" 
time="2017-08-11T21:21:16Z" level=error msg="SearchIdentities Failed with error Error Error LDAP Result Code 200 \"\": dial tcp 52.11.206.191:389: i/o timeout creating connection creating connection" 
```

**Comment by sangeethah** [comment]
Have AD auth enabled on the setup.
From within the rancher-server container, block the connection to AD ip address.

Try to log in using valid credentials.
Takes 1 minute before user is presented with ```Error communicating with authentication provider```

Debug logs:

```time="2017-08-11T21:30:25Z" level=info msg="Now generating Ldap token" 
time="2017-08-11T21:30:25Z" level=debug msg="Now creating Ldap connection" 
time="2017-08-11T21:31:25Z" level=error msg="GetToken failed with error: Error LDAP Result Code 200 \"\": dial tcp 52.11.206.191:389: i/o timeout creating connection" 
2017-08-11 21:31:25,203 ERROR [:] [] [] [] [p1020391880-445] [.a.a.i.e.ExternalServiceAuthProvider] Got error from Auth service. statusCode: 500, message: Error LDAP Result Code 200 "": dial tcp 52.11.206.191:389: i/o timeout creating connection 
```

**Comment by sangeethah** [comment]
Reopening this issue to make the AD timeout configurable .

**Comment by deniseschannon** [comment]
Available in v1.6.8-rc6

**Comment by sangeethah** [comment]
Tested with fresh install of rancher-server v1.6.8-rc6

Default Connection timeout is set to 5000 ms and it is available as configurable parameter when enabling AD access control.

From within the rancher-server container, block the connection to AD ip address
Try to log in using valid credentials.
Takes 5 seconds before user is presented with Error communicating with authentication provider
```
time="2017-08-25T00:00:11Z" level=info msg="Now generating Ldap token" 
time="2017-08-25T00:00:16Z" level=error msg="GetToken failed with error: Error creating connection: LDAP Result Code 200 \"\": dial tcp 52.11.206.191:389: i/o timeout" 
2017-08-25 00:00:16,621 ERROR [:] [] [] [] [tp1020391880-16] [.a.a.i.e.ExternalServiceAuthProvider] Got error from Auth service. statusCode: 500, message: Error creating connection: LDAP Result Code 200 "": dial tcp 52.11.206.191:389: i/o timeout 

```
Using "Edit" option in "Access Control" , change the  Connection timeout  to 10000 ms 

```

From within the rancher-server container, block the connection to AD ip address
Try to log in using valid credentials.
Takes 10 seconds before user is presented with Error communicating with authentication provider

time="2017-08-25T00:05:56Z" level=info msg="Now generating Ldap token" 
time="2017-08-25T00:06:06Z" level=error msg="GetToken failed with error: Error creating connection: LDAP Result Code 200 \"\": dial tcp 52.11.206.191:389: i/o timeout" 
2017-08-25 00:06:06,618 ERROR [:] [] [] [] [tp1020391880-16] [.a.a.i.e.ExternalServiceAuthProvider] Got error from Auth service. statusCode: 500, message: Error creating connection: LDAP Result Code 200 "": dial tcp 52.11.206.191:389: i/o timeout 

```
